cmd/kubeadm/app/apis/kubeadm/validation/validation.go

```diff
@@ -96,7 +96,6 @@ func ValidateNodeRegistrationOptions(nro *kubeadm.NodeRegistrationOptions, fldPa
 	} else {
 		allErrs = append(allErrs, apivalidation.ValidateDNS1123Subdomain(nro.Name, field.NewPath("name"))...)
 	}
-	allErrs = append(allErrs, ValidateAbsolutePath(nro.CRISocket, fldPath.Child("criSocket"))...)
```

Comment:
shouldn't it be checked that if it is non-empty and it starts with 'unix://', then the rest is valid absolute path ?

Reply:
I don't think it makes sense to do. At least in this place as it doesn't influence some areas of kubeadm, for example 'kubeadm reset' doesn't use this check.

Here is the plan so far: https://github.com/kubernetes/kubeadm/issues/926#issuecomment-398120689, but this PR is just a quick fix, making kubeadm init to be able to work with CRI.

Reply:
I'd do this:
 - In defaulting, append the `unix://` prefix if not there. This because the kubelet expects the path to be `unix://`, the path-only way is deprecated
 - In validation I'd just trim the `unix://` prefix from the string before validating it as an absolute path

Reply:
on Windows the socket protocol is at `tcp://`.

Reply:
Good note @neolit123, yeah we should source that constant from a build file using different OSes

Reply:
done, please review